core/src/repair_service.rs

```diff
@@ -518,10 +536,34 @@ impl RepairService {
         slot: Slot,
         slot_meta: &SlotMeta,
         max_repairs: usize,
+        stats: &mut RepairStats,
+        _cluster_type: ClusterType,
     ) -> Vec<ShredRepairType> {
+        const DEFER_REPAIR_THRESHOLD_MS: u64 = 300;
+        let time_delta_ms = timestamp() - slot_meta.first_shred_timestamp;
+
         if max_repairs == 0 || slot_meta.is_full() {
-            vec![]
-        } else if slot_meta.consumed == slot_meta.received {
+            return vec![];
+        }
+
+        let blockstore_highest_slot = blockstore.highest_slot().unwrap_or_default();
+        let use_time_threshold = blockstore_highest_slot
+            .map(|highest| slot >= highest.saturating_sub(1))
+            .unwrap_or_default();
+
+        error!(
+            ">>> use_time_threshold={} slot={} blockstore_highest={:?}",
+            use_time_threshold, slot, &blockstore_highest_slot
+        );
+
+        //if cluster_type == ClusterType::Testnet && time_delta_ms < DEFER_REPAIR_THRESHOLD_MS {
+        if use_time_threshold && time_delta_ms < DEFER_REPAIR_THRESHOLD_MS {
```

Comment:
Should the time threshold for missing shreds be moved from blockstore to repair? We can also bypass generating `HighestShred` repairs.